packages/server-wallet/docs/index.md

```diff
@@ -0,0 +1,183 @@
+                                                      +------------------+
+```
+
+# The responsibilities of a state channel wallet
+
+As in the [Hippocratic oath](https://en.wikipedia.org/wiki/Hippocratic_Oath): _primum non nocere,_ (first, do no harm).
+
+1. Do not lose secrets (private keys)
+2. Do not leak secrets (private keys)
+3. Store relevant states and signatures (Don’t lose any unless they are screened off by kept ones, allow states to be queried)
+4. Do not sign states or blockchain transactions unless the App grants permission
+
+Next, do some active things to protect The App’s interests.
+
+5. With implicit permission, allow an exit on chain without losing funds
+6. Detect and store blockchain challenges (for querying)
+7. Detect and store changes in channel funding (for querying)
+
+Next, do some active things to allow The App to express itself
+
+8. With explicit permission, sign updates in running application channels
+9. With implicit permission, fund channels on chain
+
+Next, allow for advanced funding relationships
+
+10. With implicit permission, sign “safe” updates in ledger channels (meaning no funds are lost)
+
+[More detail on responsibilities.](./responsibilities.md)
+
+# Objective-driven Wallet architecture
+
+In order to fulfil its responsibilities, a state channel wallet must be carefully architected. This section describes one approach that we are pursuing. Let's consider the most important responsibility of a state channels wallet, besides secure storage of secrets and states:
+
+> Signing a state or a blockchain transaction is only ever done with permission.
+
+Here, “explicit permission” shall be implied by an API call. “Implicit permission” is governed by another mechanism that we call _Objectives._ Objectives are a device that allow for
+
+- greater abstraction of the details of nitro protocol
+- greater performance and robustness of a responsible wallet
+- easier coordination between counterparties
+- short-lived request/response cycle over the API
+
+> An objective is a proposal for a goal which The App asserts that it wants to achieve.
+
+It has a _name_ and _parameters_. The parameters include a _scope_, which is a list of channelIds.
+
+> To approve an objective is to grant permission, enabling the wallet to use the private keys it stores as it sees fit, for the channels in its scope for the lifetime of the objective.
+
+> It is _not_ to give permission to spend money, only to sign states and transactions that are overall value-neutral (up to gas fees).
+
+## Implementation: the lifecycle of an objective
+
+### Creation
+
+Objectives are spawned:
+
+1. During request handling, i.e. during an API call. The API call resolves when the objective is committed to the wallet's store.
+
+Some objecties are "shared" and involve other participants. They are therefore communicated across "the wire" to other wallets. Therefore we also have objectives being spawned in a wallet:
```

Comment:
```suggestion
Some objecties are "shared" and involve other participants. They are therefore communicated across "the wire" to other wallets. Therefore, objectives can be created by the wallet itself when handling a peer message.
```

## Unrelated comment

Note that this might raise a red flag to a security-oriented engineer: this gives me the ability to DDOS your wallet and jam-pack your store with objectives.

A security engineer might suggest that The App must somehow explicitly grant permission to "create the objective", since ["allowlisting"](https://www.zdnet.com/article/uk-ncsc-to-stop-using-whitelist-and-blacklist-due-to-racial-stereotyping/) is more secure than "blocklisting"*.

\* according to some [random blog](https://blog.finjan.com/blacklisting-vs-whitelisting-understanding-the-security-benefits-of-each/)
> In fact, NIST recommends the use of whitelisting in high-risk security environments, where the integrity of individual or connected systems is critical and takes precedence over any restrictions that users might suffer in their choice or access to software.
but you can find lots of similar opinions.

(Note that there is no reference to NIST's recommendation, but they do have a [Guide to Application Whitelisting](https://csrc.nist.rip/library/alt-SP800-167.pdf).)